packages/ui/src/forum/modals/PostActionModal/DeletePostModal/DeletePostModal.tsx

```diff
@@ -21,29 +22,37 @@ import { DeletePostModalCall } from '.'
 
 export const DeletePostModal = () => {
   const {
-    modalData: { post },
+    modalData: { post, type },
     hideModal,
   } = useModal<DeletePostModalCall>()
 
   const [state, send] = useMachine(postActionMachine)
 
   const { active } = useMyMemberships()
   const { allAccounts } = useMyAccounts()
-  const { threadId, categoryId } = usePostParents(post.id)
+  const forumPostData = useForumPostParents(type === 'forum' ? post.id : '')
+  const proposalPostData = useProposalPostParents(type === 'proposal' ? post.id : '')
   const { api } = useApi()
 
-  const transaction = useMemo(
-    () =>
-      api &&
-      categoryId &&
-      threadId &&
-      api.tx.forum.deletePosts(
-        createType('ForumUserId', Number.parseInt(post.author.id)),
-        [[categoryId, threadId, post.id, true]],
-        ''
-      ),
-    [api, categoryId, threadId]
-  )
+  const transaction = useMemo(() => {
+    if (api) {
+      if (type === 'forum' && forumPostData.categoryId && forumPostData.threadId) {
+        return api.tx.forum.deletePosts(
+          createType('ForumUserId', Number.parseInt(post.author.id)),
+          [[forumPostData.categoryId, forumPostData.threadId, post.id, true]],
+          ''
+        )
+      }
+      if (type === 'proposal' && proposalPostData.threadId) {
```

Comment:
I'm thinking about this. I like the idea of not duplicating the code since there are a few places to change something. The problem I see with this solution is that it will require editing this code in the _potential_ future if we'd like to add similar actions to other places.

To combat we could introduce a way of changing the behavior from outside (let's call it dependency injection).

If that's a transaction only that needs to be updated then maybe pass it as a prop (or as a callback that returns a transaction).

Reply:
So, I've extracted this logic to PostEditor(Edit transaction) and PostContext(Delete transaction).

Reply:
I will add text check back to tests in this PR #1216